src/transformers/training_args.py

```diff
@@ -1601,13 +1601,11 @@ def __post_init__(self):
                 # no need to assert on else
 
         # if training args is specified, it will override the one specified in the accelerate config
-        if self.half_precision_backend != "apex":
```

Comment:
Here in my understanding we are not removing the `apex` backend, we are setting the `mixed_precision_dtype` variable whatever the selected backend is.

It was meant to prevent a crash in case in reach [that](https://github.com/huggingface/transformers/pull/29562/files#diff-bfceaff300c851b8e24fc50dc6638482abaec8f7d2a718e877c3828c166bcf79R1748) part of the code when the selected backend is apex (thus the variable wouldn't be initialized).

But I may be missing something here, let me know if you want it to be reverted (aside the `os.environ["ACCELERATE_MIXED_PRECISION"] = mixed_precision_dtype`) :)

Reply:
If this was preventing a crash before, why is it not crashing now?